src/libsystemd/sd-netlink/netlink-util.c

```diff
@@ -33,21 +34,35 @@ int rtnl_set_link_name(sd_netlink **rtnl, int ifindex, const char *name) {
                 if (r < 0)
                         return log_debug_errno(r, "Failed to remove '%s' from alternative names on network interface %i: %m",
                                                name, ifindex);
+
+                altname_deleted = true;
         }
 
         r = sd_rtnl_message_new_link(*rtnl, &message, RTM_SETLINK, ifindex);
         if (r < 0)
-                return r;
+                goto fail;
 
         r = sd_netlink_message_append_string(message, IFLA_IFNAME, name);
         if (r < 0)
-                return r;
+                goto fail;
 
         r = sd_netlink_call(*rtnl, message, 0, NULL);
         if (r < 0)
-                return r;
+                goto fail;
 
         return 0;
+
+fail:
+        ret = r;
+
+        if (altname_deleted) {
+                r = rtnl_set_link_alternative_names(rtnl, ifindex, STRV_MAKE(name));
```

Comment:
Not a matter, but the result of the restoring altname will be ignored, hence it is ok to introduce another variable in the `if` branch:
```c
if (altname_deleted) {
        int q = rtnl_set_link_alternative_names(...);
        if (q < 0)
                log_debug_errno(q, "...");
}
```